doc/tutorials/using_grub.rst

```diff
@@ -0,0 +1,188 @@
+.. _using_grub:
+
+Using GRUB to boot ACRN
+#######################
+`GRUB <http://www.gnu.org/software/grub/>`_ is a multiboot boot loader
+used for many popular Linux distributions. It also supports booting the
+ACRN hypervisor.
+See `<http://www.gnu.org/software/grub/grub-download.html>`_
+to get the latest GRUB source code and `<https://www.gnu.org/software/grub/grub-documentation.html>`_
+for detailed documentation.
+
+The ACRN hypervisor can boot from `multiboot protocol <http://www.gnu.org/software/grub/manual/multiboot/multiboot.html>`_
+or `multiboot2 protocol <http://www.gnu.org/software/grub/manual/multiboot2/multiboot.html>`_.
+Comparing with multiboot protocol, the multiboot2 protocol adds UEFI support.
+
+The multiboot protocol is supported by the ACRN hypervisor by default.
+When the hypervisor is loaded by GRUB's ``multiboot`` command, the
+multiboot protocol is used, and the guest kernel or ramdisk must be
+loaded with the GRUB ``module`` command.
+
+The multiboot2 protocol is supported when ``CONFIG_MULTIBOOT2`` is
+enabled in Kconfig.  When the hypervisor is loaded by GRUB's
+``multiboot2`` command, the multiboot2 protocol is used, and the guest
+kernel or ramdisk must be loaded with GRUB ``module2`` command.
+
+The ACRN hypervisor binary is built with two formats: ``acrn.32.out`` in
+ELF format and ``acrn.bin`` in RAW format. The GRUB ``multiboot``
+command support ELF format only and does not support binary relocation,
+even if ``CONFIG_RELOC`` is set. The GRUB ``multiboot2`` command support
+ELF format when ``CONFIG_RELOC`` is not set, or RAW format when
+``CONFIG_RELOC`` is set.
+
+.. note::
+   ``CONFIG_RELOC`` is set by default, so use ``acrn.32.out`` in multiboot protocol
+   and ``acrn.bin`` in multiboot2 protocol.
+
+   To make sure the platform gets the correct ACPI RSDP info, we
+   recommend using ``acrn.bin`` to load ACRN hypervisor on a UEFI platform.
+
+.. _pre-installed-grub:
+
+Using pre-installed GRUB
+************************
+
+Most Linux distributions support GRUB version 2. We can re-use
+pre-installed GRUB to load ACRN hypervisor if its version 2.04 or
+higher.
+
+Here's an example using Ubuntu to load ACRN on a scenario with two
+pre-launched VMs (the SOS_VM is also a kind of pre-launched VM):
+
+#. Copy ACRN hypervisor binary ``acrn.32.out`` (or ``acrn.bin``) and the pre-launched VM kernel images to ``/boot/``;
+
+#. Modify the ``/etc/default/grub`` file as follows to make the GRUB menu visible when booting:
+
+   .. code-block:: none
+
+      # GRUB_HIDDEN_TIMEOUT=0
+      GRUB_HIDDEN_TIMEOUT_QUIET=false
+
+#. Append the following configuration in the ``/etc/grub.d/40_custom`` file (or ``/boot/grub/custom.cfg``
```

Comment:
I think we should remove this alternative of modifying the `/boot/grub/custom.cfg` file directly.